spec/controllers/local_authorities_controller_spec.rb

```diff
@@ -63,18 +63,73 @@
   end
 
   describe "POST upload_links_csv" do
-    before { @local_authority = create(:local_authority) }
-    let(:path) { Rails.root.join("spec/lib/local-links-manager/import/fixtures/imported_links.csv") }
-    let(:csv) { Rack::Test::UploadedFile.new(path, "text/csv", true) }
-    let(:url_regex) { /http:\/\/.+\/local_authorities\/#{@local_authority.slug}/ }
+    context "with a valid CSV" do
+      before { @local_authority = create(:local_authority) }
+      let(:path) { Rails.root.join("spec/lib/local-links-manager/import/fixtures/imported_links.csv") }
+      let(:csv) { Rack::Test::UploadedFile.new(path, "text/csv", true) }
+      let(:url_regex) { /http:\/\/.+\/local_authorities\/#{@local_authority.slug}/ }
 
-    it "retrieves HTTP found" do
-      login_as_stub_user
-      post(:upload_links_csv, params: { local_authority_slug: @local_authority.slug, csv: })
+      it "retrieves HTTP found" do
+        login_as_stub_user
+        post(:upload_links_csv, params: { local_authority_slug: @local_authority.slug, csv: })
+
+        expect(response.status).to eq(302)
+        expect(response.location).to match(url_regex)
+        expect(response.headers["Content-Type"]).to eq("text/html; charset=utf-8")
+      end
+    end
 
-      expect(response.status).to eq(302)
-      expect(response.location).to match(url_regex)
-      expect(response.headers["Content-Type"]).to eq("text/html; charset=utf-8")
+    context "with errors in the CSV" do
+      before do
+        @local_authority = create(:local_authority)
+        service1 = create(:service, lgsl_code: 1)
+        service2 = create(:service, lgsl_code: 2)
+        service3 = create(:service, lgsl_code: 3)
+        service4 = create(:service, lgsl_code: 4)
+        service5 = create(:service, lgsl_code: 5)
+        service6 = create(:service, lgsl_code: 6)
+        interaction = create(:interaction, lgil_code: 1)
+        create(:service_interaction, service: service1, interaction:)
+        create(:service_interaction, service: service2, interaction:)
+        create(:service_interaction, service: service3, interaction:)
+        create(:service_interaction, service: service4, interaction:)
+        create(:service_interaction, service: service5, interaction:)
+        create(:service_interaction, service: service6, interaction:)
+      end
+
+      let(:path_all) { Rails.root.join("spec/lib/local-links-manager/import/fixtures/imported_links_all_errors.csv") }
```

Comment:
I think we only need 4 lets (csv_*) as the `path_` are only used another let rather than in any of the specs, e.g:
```
let(:csv_all) do
  path_all = Rails.root.join("spec/lib/local-links-manager/import/fixtures/imported_links_all_errors.csv")
  Rack::Test::UploadedFile.new(path_all, "text/csv", true)
end 
```

Although arguably, we probably should just include the individual files in the `it` blocks themselves as they aren't shared between tests.

Reply:
I've moved them into the tests, with a let for the base path.